In [3]:
import pandas as pd
import json
import os

In [4]:
from glob2 import glob

jsonFiles = glob('data/00-50/*.json')
print(jsonFiles)

['data/00-50\\page00.json', 'data/00-50\\page01.json', 'data/00-50\\page02.json', 'data/00-50\\page03.json', 'data/00-50\\page04.json', 'data/00-50\\page05.json', 'data/00-50\\page06.json', 'data/00-50\\page07.json', 'data/00-50\\page08.json', 'data/00-50\\page09.json', 'data/00-50\\page10.json', 'data/00-50\\page100.json', 'data/00-50\\page101.json', 'data/00-50\\page102.json', 'data/00-50\\page103.json', 'data/00-50\\page104.json', 'data/00-50\\page105.json', 'data/00-50\\page106.json', 'data/00-50\\page107.json', 'data/00-50\\page108.json', 'data/00-50\\page109.json', 'data/00-50\\page11.json', 'data/00-50\\page110.json', 'data/00-50\\page111.json', 'data/00-50\\page112.json', 'data/00-50\\page113.json', 'data/00-50\\page114.json', 'data/00-50\\page115.json', 'data/00-50\\page116.json', 'data/00-50\\page117.json', 'data/00-50\\page118.json', 'data/00-50\\page119.json', 'data/00-50\\page12.json', 'data/00-50\\page120.json', 'data/00-50\\page121.json', 'data/00-50\\page122.json', 'dat

In [5]:
from flatten_json import flatten 

dfList = []

for jsonFile in jsonFiles:
    f = open(jsonFile)
    user = json.load(f)

    df = pd.DataFrame([flatten(x) for x in user])
    dfList.append(df)
    
dfTrainingDF = pd.concat(dfList, axis=0)

In [6]:
df = dfTrainingDF 
df

name          cpf    salary    bornDate  banks_0_suitability  \
0        JOSE  99999999990  14156.55  1993-10-16                   27   
1        JOAO  99999999991   5087.48  2001-04-14                   78   
2     ANTONIO  99999999992  37402.39  1957-05-11                   55   
3   FRANCISCO  99999999993  29486.15  1959-09-28                   47   
4      CARLOS  99999999994  21993.62  1988-02-22                   22   
..        ...          ...       ...         ...                  ...   
5     ADRIANA  99999910004  18300.58  1995-01-08                   46   
6     JULIANA  99999910005  30867.57  1949-10-28                   12   
7      MARCIA  99999910006  12785.19  1988-07-08                   22   
8    FERNANDA  99999910007  35740.56  1969-01-22                   10   
9    PATRICIA  99999910008   9021.93  1965-01-22                   98   

   banks_0_startDate banks_0_institution_agency banks_0_institution_number  \
0         2016-06-26                   99999900                  999999900   
1         2019-03-23                   99999910                  999999910   
2         2016-09-09                   99999920                  999999920   
3         2019-01-28                   99999930                  999999930   
4         2016-02-28                   99999940                  999999940   
..               ...                        ...                        ...   
5         2017-03-20                   99100040                  999100040   
6         2018-05-23                   99100050                  999100050   
7         2019-03-15                   99100060                  999100060   
8         2018-12-18                   99100070                  999100070   
9         2016-05-07                   99100080                  999100080   

   banks_0_institution_bankId banks_0_institution_bankName  ...  \
0                          xp                           xp  ...   
1                          xp                           xp  ...   
2                          xp                           xp  ...   
3                          xp                           xp  ...   
4                          xp                           xp  ...   
..                        ...                          ...  ...   
5                          xp                           xp  ...   
6                          xp                           xp  ...   
7                          xp                           xp  ...   
8                          xp                           xp  ...   
9                          xp                           xp  ...   

    banks_3_checking_transactions_9_value  \
0                                     NaN   
1                                     NaN   
2                                     NaN   
3                                     NaN   
4                                     NaN   
..                                    ...   
5                                     NaN   
6                                     NaN   
7                                     NaN   
8                                     NaN   
9                                     NaN   

   banks_3_checking_transactions_9_date banks_3_checking_transactions_10_type  \
0                                   NaN                                   NaN   
1                                   NaN                                   NaN   
2                                   NaN                                   NaN   
3                                   NaN                                   NaN   
4                                   NaN                                   NaN   
..                                  ...                                   ...   
5                                   NaN                                   NaN   
6                                   NaN                                   NaN   
7                                   NaN                                   NaN   
8                                   NaN

In [7]:
df_infos = df[['name', 'cpf', 'salary', 'bornDate']]

In [8]:
df_infos

name          cpf    salary    bornDate
0        JOSE  99999999990  14156.55  1993-10-16
1        JOAO  99999999991   5087.48  2001-04-14
2     ANTONIO  99999999992  37402.39  1957-05-11
3   FRANCISCO  99999999993  29486.15  1959-09-28
4      CARLOS  99999999994  21993.62  1988-02-22
..        ...          ...       ...         ...
5     ADRIANA  99999910004  18300.58  1995-01-08
6     JULIANA  99999910005  30867.57  1949-10-28
7      MARCIA  99999910006  12785.19  1988-07-08
8    FERNANDA  99999910007  35740.56  1969-01-22
9    PATRICIA  99999910008   9021.93  1965-01-22

[1955 rows x 4 columns]

In [9]:
df_suitability = df[['banks_0_suitability',
                     'banks_1_suitability',
                     'banks_2_suitability',
                     'banks_3_suitability',]]

In [10]:
df_suitability

banks_0_suitability  banks_1_suitability  banks_2_suitability  \
0                    27                    8                    4   
1                    78                   46                   21   
2                    55                   58                   98   
3                    47                    6                   30   
4                    22                   20                   82   
..                  ...                  ...                  ...   
5                    46                   64                   31   
6                    12                   23                   15   
7                    22                   73                   71   
8                    10                   72                   54   
9                    98                   16                   42   

    banks_3_suitability  
0                    73  
1                    74  
2                    77  
3                    91  
4                    38  
..                  ...  
5                    21  
6                    95  
7                    43  
8                    33  
9                    67  

[1955 rows x 4 columns]

In [11]:
df_credit = df[[
                'banks_0_creditCard_limit', 'banks_0_creditCard_revolvingCredit', 'banks_0_creditCard_revolvingCreditTax', 'banks_0_creditCard_installmentsUsage',
                'banks_1_creditCard_limit', 'banks_1_creditCard_revolvingCredit', 'banks_1_creditCard_revolvingCreditTax', 'banks_1_creditCard_installmentsUsage',
                'banks_2_creditCard_limit', 'banks_2_creditCard_revolvingCredit', 'banks_2_creditCard_revolvingCreditTax', 'banks_2_creditCard_installmentsUsage',
                'banks_3_creditCard_limit', 'banks_3_creditCard_revolvingCredit', 'banks_3_creditCard_revolvingCreditTax', 'banks_3_creditCard_installmentsUsage',]]

In [12]:
df_credit

banks_0_creditCard_limit  banks_0_creditCard_revolvingCredit  \
0                   18936.76                             3170.00   
1                   17749.63                            26794.44   
2                   12362.14                             8700.06   
3                    5088.27                            12056.71   
4                    1079.85                            12593.90   
..                       ...                                 ...   
5                   21394.20                             2144.29   
6                    8739.70                            22788.56   
7                   24638.53                            22012.42   
8                   29565.11                             1745.48   
9                   11196.45                            27580.95   

    banks_0_creditCard_revolvingCreditTax  \
0                                0.076537   
1                                0.055231   
2                                0.031011   
3                                0.002451   
4                                0.057799   
..                                    ...   
5                                0.013714   
6                                0.006478   
7                                0.020740   
8                                0.037116   
9                                0.007847   

    banks_0_creditCard_installmentsUsage  banks_1_creditCard_limit  \
0                                   True                  13115.39   
1                                  False                   9238.97   
2                                   True                   3133.19   
3                                  False                   2818.80   
4                                  False                  25068.53   
..                                   ...                       ...   
5                                   True                  25510.18   
6                                   True                  14676.51   
7                                   True                  30497.67   
8                                  False                  23456.83   
9                                   True                  15474.85   

    banks_1_creditCard_revolvingCredit  banks_1_creditCard_revolvingCreditTax  \
0                             26550.58                               0.055107   
1                             27216.01                               0.032354   
2                             22507.72                               0.016986   
3                             22381.72                               0.037835   
4                             25905.59                               0.049948   
..                                 ...                                    ...   
5                             27244.07                               0.006887   
6                             23964.83                               0.039753   
7                             18442.45                               0.011280   
8                              6607.46                               0.033763   
9                              6043.28                               0.042206   

    banks_1_creditCard_installmentsUsage  banks_2_creditCard_limit  \
0                                   True                  19777.52   
1                                  False                   3252.09   
2                                   True                  20466.04   
3                                   True                  10242.37   
4                                  False                  15479.99   
..                                   ...                       ...   
5                                  False                  22116.20   
6                                   True                  16296.29   
7                                  False                  30249.40   
8                                  False                   7317.77   
9                                   True                  2330

In [13]:
df_savings = df[[
                 'banks_0_saving_balance',
                 'banks_1_saving_balance',
                 'banks_2_saving_balance',
                 'banks_3_saving_balance',]]

In [14]:
df_savings

banks_0_saving_balance  banks_1_saving_balance  banks_2_saving_balance  \
0                        0                       0                 1629454   
1                        0                       0                  775788   
2                        0                       0                       0   
3                   411693                       0                 2501405   
4                  1498482                       0                 2986233   
..                     ...                     ...                     ...   
5                        0                       0                  304051   
6                        0                  422317                       0   
7                  1328351                       0                       0   
8                        0                 2590866                       0   
9                        0                 1483618                       0   

    banks_3_saving_balance  
0                  2747833  
1                  1018272  
2                        0  
3                  2301785  
4                   344070  
..                     ...  
5                        0  
6                  1057971  
7                  2147581  
8                        0  
9                  2841650  

[1955 rows x 4 columns]

In [15]:
df_pix_0 = df[['banks_0_pixHistory_0_value', 
               'banks_0_pixHistory_1_value',
               'banks_0_pixHistory_2_value',
               'banks_0_pixHistory_3_value',
               'banks_0_pixHistory_4_value',
               'banks_0_pixHistory_5_value',
               'banks_0_pixHistory_6_value',
               'banks_0_pixHistory_7_value',
               'banks_0_pixHistory_8_value',
               'banks_0_pixHistory_9_value']]

df['banks_0_pix_volume'] = df_pix_0.sum(axis=1)

df_pix_1 = df[['banks_1_pixHistory_0_value', 
               'banks_1_pixHistory_1_value',
               'banks_1_pixHistory_2_value',
               'banks_1_pixHistory_3_value',
               'banks_1_pixHistory_4_value',
               'banks_1_pixHistory_5_value',
               'banks_1_pixHistory_6_value',
               'banks_1_pixHistory_7_value',
               'banks_1_pixHistory_8_value',
               'banks_1_pixHistory_9_value']]

df['banks_1_pix_volume'] = df_pix_1.sum(axis=1)

df_pix_2 = df[['banks_2_pixHistory_0_value', 
               'banks_2_pixHistory_1_value',
               'banks_2_pixHistory_2_value',
               'banks_2_pixHistory_3_value',
               'banks_2_pixHistory_4_value',
               'banks_2_pixHistory_5_value',
               'banks_2_pixHistory_6_value',
               'banks_2_pixHistory_7_value',
               'banks_2_pixHistory_8_value',
               'banks_2_pixHistory_9_value']]

df['banks_2_pix_volume'] = df_pix_2.sum(axis=1)

df_pix_3 = df[['banks_3_pixHistory_0_value', 
               'banks_3_pixHistory_1_value',
               'banks_3_pixHistory_2_value',
               'banks_3_pixHistory_3_value',
               'banks_3_pixHistory_4_value',
               'banks_3_pixHistory_5_value',
               'banks_3_pixHistory_6_value',
               'banks_3_pixHistory_7_value',
               'banks_3_pixHistory_8_value',
               'banks_3_pixHistory_9_value']]

df['banks_3_pix_volume'] = df_pix_3.sum(axis=1)

In [16]:
df_pix = df[[
             'banks_0_pix_volume', 'banks_1_pix_volume', 'banks_2_pix_volume','banks_3_pix_volume']]

In [17]:
df_pix

banks_0_pix_volume  banks_1_pix_volume  banks_2_pix_volume  \
0            597657.25           528224.43           619350.86   
1            363973.39           547951.79           558851.77   
2            503961.19           435122.01           687595.30   
3            391737.62           512708.33           513257.90   
4            524346.85           500632.11           607713.84   
..                 ...                 ...                 ...   
5            385978.67           385893.87           585015.14   
6            297791.13           516260.30           385063.48   
7            622439.64           584298.54           684094.53   
8            462443.33           439526.04           369704.07   
9            518040.63           555089.64           533048.46   

    banks_3_pix_volume  
0            485338.28  
1            531797.88  
2            547721.87  
3            252417.95  
4            482697.51  
..                 ...  
5            497168.79  
6            758423.02  
7            538012.05  
8            561292.68  
9            519346.83  

[1955 rows x 4 columns]

In [18]:
df_lines = df[[
               'banks_0_consumedCreditLines_0_type', 'banks_0_consumedCreditLines_0_tax', 'banks_0_consumedCreditLines_0_installments',
               'banks_1_consumedCreditLines_0_type', 'banks_1_consumedCreditLines_0_tax', 'banks_1_consumedCreditLines_0_installments',
               'banks_2_consumedCreditLines_0_type', 'banks_2_consumedCreditLines_0_tax', 'banks_2_consumedCreditLines_0_installments',
               'banks_3_consumedCreditLines_0_type', 'banks_3_consumedCreditLines_0_tax', 'banks_3_consumedCreditLines_0_installments']]

In [19]:
df_lines

banks_0_consumedCreditLines_0_type  banks_0_consumedCreditLines_0_tax  \
0                             leasing                           0.432114   
1                       personal-loan                           0.697097   
2                       personal-loan                           0.625389   
3                             leasing                           0.054133   
4                             leasing                           0.346786   
..                                ...                                ...   
5                             leasing                           0.219908   
6                       personal-loan                           0.631160   
7                       personal-loan                           0.057862   
8                       personal-loan                           0.658566   
9                             leasing                           0.545414   

    banks_0_consumedCreditLines_0_installments  \
0                                           44   
1                                           36   
2                                           58   
3                                           26   
4                                           56   
..                                         ...   
5                                           13   
6                                           36   
7                                           25   
8                                           50   
9                                           31   

   banks_1_consumedCreditLines_0_type  banks_1_consumedCreditLines_0_tax  \
0                             leasing                           0.978549   
1                       personal-loan                           0.239326   
2                       personal-loan                           0.974277   
3                       personal-loan                           0.613321   
4                       personal-loan                           0.990649   
..                                ...                                ...   
5                       personal-loan                           0.150455   
6                             leasing                           0.449917   
7                       personal-loan                           0.069725   
8                       personal-loan                           0.882166   
9                             leasing                           0.352224   

    banks_1_consumedCreditLines_0_installments  \
0                                           54   
1                                            6   
2                                            2   
3                                           36   
4                                           55   
..                                         ...   
5                                           20   
6                                           30   
7                                            6   
8                                           57   
9                                           57   

   banks_2_consumedCreditLines_0_type  banks_2_consumedCreditLines_0_tax  \
0                             leasing                           0.306193   
1                       personal-loan                           0.318732   
2                             leasing                           0.633576   
3                             leasing                           0.381224   
4                       personal-loan                           0.093006   
..                                ...                                ...   
5                       personal-loan                           0.398112   
6                             leasing                           0.520737   
7                       personal-loan                           0.180821   
8                             leasing                           0.250957   
9                       personal-loan                           0.931598   

    banks_2_consumedCreditLines_0_installments  \
0           

In [20]:
df_stocks_0_vol = df[['banks_0_investments_stocks_0_volumn', 'banks_0_investments_stocks_1_volumn', 'banks_0_investments_stocks_2_volumn', 'banks_0_investments_stocks_3_volumn']]
df_stocks_0_val = df[['banks_0_investments_stocks_0_value', 'banks_0_investments_stocks_1_value', 'banks_0_investments_stocks_2_value', 'banks_0_investments_stocks_3_value']]

df_stocks = pd.DataFrame()

df_stocks['df_stocks_0_total_bank0'] = df_stocks_0_vol['banks_0_investments_stocks_0_volumn']*df_stocks_0_val['banks_0_investments_stocks_0_value']
df_stocks['df_stocks_1_total_bank0'] = df_stocks_0_vol['banks_0_investments_stocks_1_volumn']*df_stocks_0_val['banks_0_investments_stocks_1_value']
df_stocks['df_stocks_2_total_bank0'] = df_stocks_0_vol['banks_0_investments_stocks_2_volumn']*df_stocks_0_val['banks_0_investments_stocks_2_value']
df_stocks['df_stocks_3_total_bank0'] = df_stocks_0_vol['banks_0_investments_stocks_3_volumn']*df_stocks_0_val['banks_0_investments_stocks_3_value']

df['banks_0_stocks_volume'] = df_stocks.sum(axis=1)

df_stocks_1_vol = df[['banks_1_investments_stocks_0_volumn', 'banks_1_investments_stocks_1_volumn', 'banks_1_investments_stocks_2_volumn', 'banks_1_investments_stocks_3_volumn']]
df_stocks_1_val = df[['banks_1_investments_stocks_0_value', 'banks_1_investments_stocks_1_value', 'banks_1_investments_stocks_2_value', 'banks_1_investments_stocks_3_value']]

df_stocks['df_stocks_0_total_bank1'] = df_stocks_1_vol['banks_1_investments_stocks_0_volumn']*df_stocks_1_val['banks_1_investments_stocks_0_value']
df_stocks['df_stocks_1_total_bank1'] = df_stocks_1_vol['banks_1_investments_stocks_1_volumn']*df_stocks_1_val['banks_1_investments_stocks_1_value']
df_stocks['df_stocks_2_total_bank1'] = df_stocks_1_vol['banks_1_investments_stocks_2_volumn']*df_stocks_1_val['banks_1_investments_stocks_2_value']
df_stocks['df_stocks_3_total_bank1'] = df_stocks_1_vol['banks_1_investments_stocks_3_volumn']*df_stocks_1_val['banks_1_investments_stocks_3_value']

df['banks_1_stocks_volume'] = df_stocks.sum(axis=1)

df_stocks_2_vol = df[['banks_2_investments_stocks_0_volumn', 'banks_2_investments_stocks_1_volumn', 'banks_2_investments_stocks_2_volumn', 'banks_2_investments_stocks_3_volumn']]
df_stocks_2_val = df[['banks_2_investments_stocks_0_value', 'banks_2_investments_stocks_1_value', 'banks_2_investments_stocks_2_value', 'banks_2_investments_stocks_3_value']]

df_stocks['df_stocks_0_total_bank2'] = df_stocks_2_vol['banks_2_investments_stocks_0_volumn']*df_stocks_2_val['banks_2_investments_stocks_0_value']
df_stocks['df_stocks_1_total_bank2'] = df_stocks_2_vol['banks_2_investments_stocks_1_volumn']*df_stocks_2_val['banks_2_investments_stocks_1_value']
df_stocks['df_stocks_2_total_bank2'] = df_stocks_2_vol['banks_2_investments_stocks_2_volumn']*df_stocks_2_val['banks_2_investments_stocks_2_value']
df_stocks['df_stocks_3_total_bank2'] = df_stocks_2_vol['banks_2_investments_stocks_3_volumn']*df_stocks_2_val['banks_2_investments_stocks_3_value']

df['banks_2_stocks_volume'] = df_stocks.sum(axis=1)

df_stocks_3_vol = df[['banks_3_investments_stocks_0_volumn', 'banks_3_investments_stocks_1_volumn', 'banks_3_investments_stocks_2_volumn', 'banks_3_investments_stocks_3_volumn']]
df_stocks_3_val = df[['banks_3_investments_stocks_0_value', 'banks_3_investments_stocks_1_value', 'banks_3_investments_stocks_2_value', 'banks_3_investments_stocks_3_value']]

df_stocks['df_stocks_0_total_bank2'] = df_stocks_3_vol['banks_3_investments_stocks_0_volumn']*df_stocks_3_val['banks_3_investments_stocks_0_value']
df_stocks['df_stocks_1_total_bank2'] = df_stocks_3_vol['banks_3_investments_stocks_1_volumn']*df_stocks_3_val['banks_3_investments_stocks_1_value']
df_stocks['df_stocks_2_total_bank2'] = df_stocks_3_vol['banks_3_investments_stocks_2_volumn']*df_stocks_3_val['banks_3_investments_stocks_2_value']
df_stocks['df_stocks_3_total_bank2'] = df_stocks_3_vol['banks_3_investments_stocks_3_volumn']*df_stocks_3_val['banks_3_investments_stocks_3_value']

df['banks_3_stocks_volume'] = df_stocks.sum(axis=1)


In [21]:
df_stocks_final = df[['banks_0_stocks_volume', 'banks_1_stocks_volume', 'banks_2_stocks_volume','banks_3_stocks_volume']]

In [22]:
df_stocks_final

banks_0_stocks_volume  banks_1_stocks_volume  banks_2_stocks_volume  \
0                  902816                1840768                3160305   
1                  847479                2028985                2690246   
2                  797500                1560536                2042604   
3                 1470359                2459673                4094993   
4                 1104848                1625381                2725231   
..                    ...                    ...                    ...   
5                 1360709                3231699                3810281   
6                 1485673                3700800                4432373   
7                 1018395                2346399                3870369   
8                  458941                 827701                1373250   
9                  763176                1920425                3586956   

    banks_3_stocks_volume  
0                 2850637  
1                 3786698  
2                 2270493  
3                 3108295  
4                 2825460  
..                    ...  
5                 4590675  
6                 5382368  
7                 3142210  
8                 2543678  
9                 2929619  

[1955 rows x 4 columns]

In [23]:
df_fundos = pd.DataFrame()

df_fundos_0_val = df[['banks_0_investments_investmentFunds_0_value', 'banks_0_investments_investmentFunds_1_value', 'banks_0_investments_investmentFunds_2_value', 'banks_0_investments_investmentFunds_3_value']]
df_fundos_0_vol = df[['banks_0_investments_investmentFunds_0_volumn', 'banks_0_investments_investmentFunds_1_volumn', 'banks_0_investments_investmentFunds_2_volumn', 'banks_0_investments_investmentFunds_3_volumn']]

df_fundos['df_fundos_0_total_bank0'] = df_fundos_0_val['banks_0_investments_investmentFunds_0_value']*df_fundos_0_vol['banks_0_investments_investmentFunds_0_volumn']
df_fundos['df_fundos_1_total_bank0'] = df_fundos_0_val['banks_0_investments_investmentFunds_1_value']*df_fundos_0_vol['banks_0_investments_investmentFunds_1_volumn']
df_fundos['df_fundos_2_total_bank0'] = df_fundos_0_val['banks_0_investments_investmentFunds_2_value']*df_fundos_0_vol['banks_0_investments_investmentFunds_2_volumn']
df_fundos['df_fundos_3_total_bank0'] = df_fundos_0_val['banks_0_investments_investmentFunds_3_value']*df_fundos_0_vol['banks_0_investments_investmentFunds_3_volumn']

df['banks_0_fundos_volume'] = df_fundos.sum(axis=1)

df_fundos_0_val = df[['banks_1_investments_investmentFunds_0_value', 'banks_1_investments_investmentFunds_1_value', 'banks_1_investments_investmentFunds_2_value', 'banks_1_investments_investmentFunds_3_value']]
df_fundos_0_vol = df[['banks_1_investments_investmentFunds_0_volumn', 'banks_1_investments_investmentFunds_1_volumn', 'banks_1_investments_investmentFunds_2_volumn', 'banks_1_investments_investmentFunds_3_volumn']]

df_fundos['df_fundos_0_total_bank1'] = df_fundos_0_val['banks_1_investments_investmentFunds_0_value']*df_fundos_0_vol['banks_1_investments_investmentFunds_0_volumn']
df_fundos['df_fundos_1_total_bank1'] = df_fundos_0_val['banks_1_investments_investmentFunds_1_value']*df_fundos_0_vol['banks_1_investments_investmentFunds_1_volumn']
df_fundos['df_fundos_2_total_bank1'] = df_fundos_0_val['banks_1_investments_investmentFunds_2_value']*df_fundos_0_vol['banks_1_investments_investmentFunds_2_volumn']
df_fundos['df_fundos_3_total_bank1'] = df_fundos_0_val['banks_1_investments_investmentFunds_3_value']*df_fundos_0_vol['banks_1_investments_investmentFunds_3_volumn']

df['banks_1_fundos_volume'] = df_fundos.sum(axis=1)

df_fundos_0_val = df[['banks_2_investments_investmentFunds_0_value', 'banks_2_investments_investmentFunds_1_value', 'banks_2_investments_investmentFunds_2_value', 'banks_2_investments_investmentFunds_3_value']]
df_fundos_0_vol = df[['banks_2_investments_investmentFunds_0_volumn', 'banks_2_investments_investmentFunds_1_volumn', 'banks_2_investments_investmentFunds_2_volumn', 'banks_2_investments_investmentFunds_3_volumn']]

df_fundos['df_fundos_0_total_bank2'] = df_fundos_0_val['banks_2_investments_investmentFunds_0_value']*df_fundos_0_vol['banks_2_investments_investmentFunds_0_volumn']
df_fundos['df_fundos_1_total_bank2'] = df_fundos_0_val['banks_2_investments_investmentFunds_1_value']*df_fundos_0_vol['banks_2_investments_investmentFunds_1_volumn']
df_fundos['df_fundos_2_total_bank2'] = df_fundos_0_val['banks_2_investments_investmentFunds_2_value']*df_fundos_0_vol['banks_2_investments_investmentFunds_2_volumn']
df_fundos['df_fundos_3_total_bank2'] = df_fundos_0_val['banks_2_investments_investmentFunds_3_value']*df_fundos_0_vol['banks_2_investments_investmentFunds_3_volumn']

df['banks_2_fundos_volume'] = df_fundos.sum(axis=1)

df_fundos_0_val = df[['banks_3_investments_investmentFunds_0_value', 'banks_3_investments_investmentFunds_1_value', 'banks_3_investments_investmentFunds_2_value', 'banks_3_investments_investmentFunds_3_value']]
df_fundos_0_vol = df[['banks_3_investments_investmentFunds_0_volumn', 'banks_3_investments_investmentFunds_1_volumn', 'banks_3_investments_investmentFunds_2_volumn', 'banks_3_investments_investmentFunds_3_volumn']]

df_fundos['df_fundos_0_total_bank3'] = df_fundos_0_val['banks_3_investments_investmentFunds_0_value']*df_fundos_0_vol['banks_3_investments_investmentFunds_0_volumn']
df_fundos['df_fundos_1_total_bank3'] = df_fundos_0_val['banks_3_investments_investmentFunds_1_value']*df_fundos_0_vol['banks_3_investments_investmentFunds_1_volumn']
df_fundos['df_fundos_2_total_bank3'] = df_fundos_0_val['banks_3_investments_investmentFunds_2_value']*df_fundos_0_vol['banks_3_investments_investmentFunds_2_volumn']
df_fundos['df_fundos_3_total_bank3'] = df_fundos_0_val['banks_3_investments_investmentFunds_3_value']*df_fundos_0_vol['banks_3_investments_investmentFunds_3_volumn']

df['banks_3_fundos_volume'] = df_fundos.sum(axis=1)


In [24]:
df_fundos_final = df[['banks_0_fundos_volume', 'banks_1_fundos_volume', 'banks_2_fundos_volume','banks_3_fundos_volume']]

In [25]:
df_fundos_final

banks_0_fundos_volume  banks_1_fundos_volume  banks_2_fundos_volume  \
0                 1439037                4637670                5705326   
1                  921824                2786392                4504878   
2                 2336590                4009424                6931627   
3                  506489                2580986                4560772   
4                 1693940                4298609                6917668   
..                    ...                    ...                    ...   
5                 2566152                3021042                5117182   
6                  930711                2623616                4325937   
7                 1647554                3716490                5027688   
8                 1760847                4064790                6955594   
9                 1723562                4605985                6960122   

    banks_3_fundos_volume  
0                 6107081  
1                 5344946  
2                 8360838  
3                 7452572  
4                 7917787  
..                    ...  
5                 5972323  
6                 6759435  
7                 8288933  
8                 8485555  
9                 9250414  

[1955 rows x 4 columns]

In [26]:
df_final = pd.concat([df_infos, df_suitability, df_credit, df_savings, df_pix, df_lines, df_stocks_final, df_fundos_final], axis=1)
df_final.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1955 entries, 0 to 9
Data columns (total 52 columns):
 #   Column                                      Non-Null Count  Dtype  
---  ------                                      --------------  -----  
 0   name                                        1955 non-null   object 
 1   cpf                                         1955 non-null   object 
 2   salary                                      1955 non-null   float64
 3   bornDate                                    1955 non-null   object 
 4   banks_0_suitability                         1955 non-null   int64  
 5   banks_1_suitability                         1955 non-null   int64  
 6   banks_2_suitability                         1955 non-null   int64  
 7   banks_3_suitability                         1955 non-null   int64  
 8   banks_0_creditCard_limit                    1955 non-null   float64
 9   banks_0_creditCard_revolvingCredit          1955 non-null   float64
 10  banks_0_creditC

In [27]:
df_bank_0 = pd.concat([df_infos, df_final.loc[:,df_final.columns.str.startswith('banks_0')]], axis=1)
df_bank_1 = pd.concat([df_infos, df_final.loc[:,df_final.columns.str.startswith('banks_1')]], axis=1)
df_bank_2 = pd.concat([df_infos, df_final.loc[:,df_final.columns.str.startswith('banks_2')]], axis=1)
df_bank_3 = pd.concat([df_infos, df_final.loc[:,df_final.columns.str.startswith('banks_3')]], axis=1)

In [28]:
df_bank_0.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1955 entries, 0 to 9
Data columns (total 16 columns):
 #   Column                                      Non-Null Count  Dtype  
---  ------                                      --------------  -----  
 0   name                                        1955 non-null   object 
 1   cpf                                         1955 non-null   object 
 2   salary                                      1955 non-null   float64
 3   bornDate                                    1955 non-null   object 
 4   banks_0_suitability                         1955 non-null   int64  
 5   banks_0_creditCard_limit                    1955 non-null   float64
 6   banks_0_creditCard_revolvingCredit          1955 non-null   float64
 7   banks_0_creditCard_revolvingCreditTax       1955 non-null   float64
 8   banks_0_creditCard_installmentsUsage        1955 non-null   bool   
 9   banks_0_saving_balance                      1955 non-null   int64  
 10  banks_0_pix_vol

Banco 0

In [29]:
corr0 = df_bank_0.corr()
corr0.style.background_gradient(cmap='coolwarm')

In [30]:
df_bank_0.corr().unstack().sort_values().drop_duplicates()

banks_0_suitability       salary                                  -0.147257
banks_0_fundos_volume     banks_0_creditCard_revolvingCreditTax   -0.125780
banks_0_stocks_volume     banks_0_consumedCreditLines_0_tax       -0.116826
salary                    banks_0_creditCard_installmentsUsage    -0.115277
banks_0_creditCard_limit  banks_0_pix_volume                      -0.101963
                                                                     ...   
banks_0_pix_volume        banks_0_creditCard_revolvingCreditTax    0.087499
banks_0_saving_balance    banks_0_creditCard_revolvingCredit       0.094551
                          banks_0_creditCard_revolvingCreditTax    0.099976
banks_0_fundos_volume     salary                                   0.116480
salary                    salary                                   1.000000
Length: 67, dtype: float64

Banco 1 

In [31]:
corr1 = df_bank_1.corr()
corr1.style.background_gradient(cmap='coolwarm')

In [32]:
df_bank_1.corr().unstack().sort_values().drop_duplicates()

banks_1_fundos_volume                       banks_1_saving_balance                       -0.194627
banks_1_creditCard_revolvingCreditTax       banks_1_pix_volume                           -0.155081
banks_1_consumedCreditLines_0_installments  banks_1_saving_balance                       -0.134225
banks_1_suitability                         banks_1_saving_balance                       -0.112213
banks_1_creditCard_revolvingCreditTax       banks_1_consumedCreditLines_0_installments   -0.101928
                                                                                            ...   
banks_1_consumedCreditLines_0_tax           salary                                        0.122995
banks_1_pix_volume                          banks_1_consumedCreditLines_0_tax             0.124379
salary                                      banks_1_stocks_volume                         0.126714
banks_1_creditCard_installmentsUsage        banks_1_consumedCreditLines_0_installments    0.146597
salary    

Banco 2

In [33]:
corr2 = df_bank_2.corr()
corr2.style.background_gradient(cmap='coolwarm')

In [34]:
df_bank_2.corr().unstack().sort_values().drop_duplicates()

banks_2_creditCard_revolvingCreditTax  banks_2_creditCard_revolvingCredit      -0.189292
banks_2_consumedCreditLines_0_tax      banks_2_creditCard_revolvingCredit      -0.168630
banks_2_creditCard_installmentsUsage   banks_2_fundos_volume                   -0.132203
banks_2_creditCard_revolvingCreditTax  banks_2_saving_balance                  -0.117722
                                       banks_2_stocks_volume                   -0.107119
                                                                                  ...   
banks_2_pix_volume                     banks_2_creditCard_installmentsUsage     0.102946
salary                                 banks_2_fundos_volume                    0.109576
banks_2_creditCard_limit               banks_2_creditCard_revolvingCreditTax    0.138475
banks_2_creditCard_installmentsUsage   banks_2_suitability                      0.148207
salary                                 salary                                   1.000000
Length: 67, dtype: fl

Banco 3 

In [35]:
corr3 = df_bank_3.corr()
corr3.style.background_gradient(cmap='coolwarm')

In [36]:
df_bank_3.corr().unstack().sort_values().drop_duplicates()

banks_3_pix_volume                          banks_3_fundos_volume                        -0.189349
banks_3_consumedCreditLines_0_tax           banks_3_creditCard_revolvingCreditTax        -0.124749
banks_3_pix_volume                          banks_3_saving_balance                       -0.111293
banks_3_consumedCreditLines_0_installments  banks_3_saving_balance                       -0.110155
banks_3_creditCard_revolvingCredit          banks_3_saving_balance                       -0.087556
                                                                                            ...   
banks_3_stocks_volume                       salary                                        0.141944
banks_3_fundos_volume                       banks_3_saving_balance                        0.152558
banks_3_stocks_volume                       banks_3_consumedCreditLines_0_installments    0.156361
banks_3_consumedCreditLines_0_tax           banks_3_creditCard_installmentsUsage          0.176001
salary    

## features mais correlacionadas
### banco 0 
banks_0_pix_volume        banks_0_creditCard_revolvingCreditTax    0.087499
banks_0_saving_balance    banks_0_creditCard_revolvingCredit       0.094551
                          banks_0_creditCard_revolvingCreditTax    0.099976
banks_0_fundos_volume     salary                                   0.116480

### banco 1 
bbanks_1_consumedCreditLines_0_tax           salary                                        0.122995
banks_1_pix_volume                          banks_1_consumedCreditLines_0_tax             0.124379
salary                                      banks_1_stocks_volume                         0.126714
banks_1_creditCard_installmentsUsage        banks_1_consumedCreditLines_0_installments    0.146597

### banco 2 
banks_2_pix_volume                     banks_2_creditCard_installmentsUsage     0.102946
salary                                 banks_2_fundos_volume                    0.109576
banks_2_creditCard_limit               banks_2_creditCard_revolvingCreditTax    0.138475
banks_2_creditCard_installmentsUsage   banks_2_suitability                      0.148207

### banco 3 
banks_3_stocks_volume                       salary                                        0.141944
banks_3_fundos_volume                       banks_3_saving_balance                        0.152558
banks_3_stocks_volume                       banks_3_consumedCreditLines_0_installments    0.156361
banks_3_consumedCreditLines_0_tax           banks_3_creditCard_installmentsUsage          0.176001


In [37]:
from pycaret.classification import *

s = setup(df_bank_0, target = 'banks_0_fundos_volume')

In [38]:
stop

NameError: name 'stop' is not defined

In [ ]:
best_model = compare_models(sort='AUC')
print(best_model)

IntProgress(value=0, description='Processing: ', max=79)

Initiated  . . . . . . . . . . . . . . . . . .                      23:14:52
Status     . . . . . . . . . . . . . . . . . .              Fitting 10 Folds
Estimator  . . . . . . . . . . . . . . . . . .  Gradient Boosting Classifier

In [42]:
xgboost = create_model('xgboost')

IntProgress(value=0, description='Processing: ', max=4)

Initiated  . . . . . . . . . . . . . . . . . .                   23:48:46
Status     . . . . . . . . . . . . . . . . . .           Fitting 10 Folds
Estimator  . . . . . . . . . . . . . . . . . .  Extreme Gradient Boosting

Empty DataFrame
Columns: [Accuracy, AUC, Recall, Prec., F1, Kappa, MCC]
Index: []

KeyboardInterrupt: 